internal/pkg/scaffold/olm-catalog/csv_updaters.go

```diff
@@ -138,6 +155,47 @@ func (store *updaterStore) AddDeploymentSpec(yamlDoc []byte) error {
 	return nil
 }
 
+// setWatchNamespacesEnv sets WATCH_NAMESPACE to olmTNString in dep if
+// WATCH_NAMESPACE exists in a pod spec container's env list.
+func setWatchNamespacesEnv(dep *appsv1.Deployment) {
+	overwriteContainerEnvVar(dep, k8sutil.WatchNamespaceEnvVar, newEnvVar(k8sutil.WatchNamespaceEnvVar, olmTNMeta))
+}
+
+func overwriteContainerEnvVar(dep *appsv1.Deployment, name string, ev corev1.EnvVar) {
+	for _, c := range dep.Spec.Template.Spec.Containers {
+		for i := 0; i < len(c.Env); i++ {
+			if c.Env[i].Name == name {
+				c.Env[i] = ev
+			}
+		}
+	}
+}
+
+func newEnvVar(name, fieldPath string) corev1.EnvVar {
+	return corev1.EnvVar{
+		Name: name,
+		ValueFrom: &corev1.EnvVarSource{
+			FieldRef: &corev1.ObjectFieldSelector{
+				FieldPath: fieldPath,
+			},
+		},
+	}
+}
+
+// OLM places the set of target namespaces for the operator in
+// "metadata.annotations['olm.targetNamespaces']". This value should be
+// referenced in either:
+//	- The Deployment's pod spec WATCH_NAMESPACE env variable.
+//	- Some other Deployment pod spec field.
+func depHasOLMNamespaces(dep *appsv1.Deployment) bool {
```

Comment:
What happens when a user uses a StatefulSet or less likely a Pod to deploy their operator?

Reply:
I think OLM only lets you install your operator via a Deployment in the CSV's install. And this would just be checking the deployment for the `spec.install.deployment` field of the CSV.
https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/building-your-csv.md#operator-install